### PR TITLE
helm: Set default second factor to 'otp'

### PR DESCRIPTION
**Pull request by webvictim**
https://github.com/gravitational/teleport/pull/9877 set the default for `auth_service.authentication.second_factor` to `on`, which causes an error when a webauthn configuration is not also supplied: `ERROR: missing required webauthn configuration for second factor type "on"`

The actual Teleport default is `otp`, which should be respected in the chart values.

This also adds the potential to set `off`, which is itself a valid option.

### CODE REVIEW

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 1]
```diff
@@ -73,9 +73,11 @@ This reference details available values for the `teleport-cluster` chart.
 
 | Type | Default value | Required? | `teleport.yaml` equivalent | Can be used in `custom` mode? |
 | - | - | - | - | - |
-| `string` | `on` | Yes | `auth_service.authentication.second_factor` | ❌ |
+| `string` | `otp` | Yes | `auth_service.authentication.second_factor` | ❌ |
 
-`authenticationSecondFactor.secondFactor` controls the second factor used for local user authentication. Possible values are `on`, `optional` and `webauthn`.
+`authenticationSecondFactor.secondFactor` controls the second factor used for local user authentication. Possible values are `off` (not recommended), `on`, `otp`, `optional` and `webauthn`.
+
+When set to `on` or `webauthn`, the `authenticationSecondFactor.webauthn` section must also be set.
```
Comment (Joerger) [inline docs/pages/kubernetes-access/helm/reference.mdx:80]: This more accurately represents the functionality.
```suggestion
When set to `on` or `optional`, the secondFactor used will default to `webauthn` or `u2f`, depending on whether `authenticationSecondFactor.webauthn` or `authenticationSeconfFactor.u2f` is set.
```
If the issue is that you set `on` and set the `u2f` field and still got the error log, then this must be a bug.

Comment (zmb3) [reply docs/pages/kubernetes-access/helm/reference.mdx:80]: For v8 and v9, yes. But on master we've dropped u2f support.

**Review by zmb3 — APPROVED** [review]

File: docs/pages/kubernetes-access/helm/reference.mdx (hunk 1 above)
Comment (codingllama) [reply docs/pages/kubernetes-access/helm/reference.mdx]: @Joerger is correct, we are missing "optional" in the list. It functions rather similarly to "on" in terms of configuration.

I don't see an U2F section in the schema, did we ever support it in the chart? U2F settings are relevant for backwards compatibility.

It also looks like we are setting the cluster name as the WebAuthn RPID? In this case, shouldn't the webauthn section always be present?

Comment (webvictim) [reply docs/pages/kubernetes-access/helm/reference.mdx:80]: Got it. Given that I'm only intending to backport this to v8 and v9 (which both have decent webauthn support) and there's no support for `authenticationSecondFactor.u2f` in the Helm chart, I don't think I'll mention `u2f` here at all for ease.

I do think I should add `optional` to the list of values which require the `webauthn` section to be set, though.

**Review by ptgott — APPROVED** [review]

File: docs/pages/kubernetes-access/helm/reference.mdx (hunk 1 above)
Comment (webvictim) [reply docs/pages/kubernetes-access/helm/reference.mdx]: Also, I understand your comments about `u2f` but I'm loath to introduce another full section and associated Helm parsing/validation for a standard which I think we now consider deprecated in favour of webauthn. Any existing Helm chart users who wanted U2F are likely to be deploying a custom Teleport config which enables it, so I think we can leave that as-is.

**Review by Joerger — APPROVED** [review]

File: docs/pages/kubernetes-access/helm/reference.mdx (hunk 1 above)
Comment (codingllama) [reply docs/pages/kubernetes-access/helm/reference.mdx]: > we now always write rp_id when second factor is not set to ...

File: examples/chart/teleport-cluster/templates/config.yaml [hunk 2]
```diff
@@ -59,16 +59,18 @@ data:
     {{- if .Values.authenticationSecondFactor.secondFactor }}
         second_factor: {{ .Values.authenticationSecondFactor.secondFactor }}
     {{- end }}
-    {{- if .Values.authenticationSecondFactor.webauthn }}
+    {{- if not (or (eq .Values.authenticationSecondFactor.secondFactor "off") (eq .Values.authenticationSecondFactor.secondFactor "otp")) }}
```
Comment (codingllama) [inline examples/chart/teleport-cluster/templates/config.yaml:62]: Drop this logic and always write it? It won't hurt.

**Review by codingllama — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks a lot for addressing this, @webvictim.

File: examples/chart/teleport-cluster/templates/config.yaml (hunk 2 above)
Comment (webvictim) [reply examples/chart/teleport-cluster/templates/config.yaml:62]: I considered this, but I think it'll just confuse people if they see that section appearing in their configs when second factor is disabled. The logic is a little more brittle but hopefully produces a better end result.

**Pull request merged**